src/main/setupInvestHandlers.js

```diff
@@ -127,31 +90,38 @@ export function setupInvestRunHandlers(investExe) {
 
     // There's no general way to know that a spawned process started,
     // so this logic to listen once on stdout seems like the way.
-    investRun.stdout.once('data', async (data) => {
-      const logfile = await findMostRecentLogfile(args.workspace_dir);
-      // TODO: handle case when logfile is still undefined?
-      // Could be if some stdout is emitted before a logfile exists.
-      logger.debug(`invest logging to: ${logfile}`);
-      runningJobs[args.workspace_dir] = investRun.pid;
-      event.reply(`invest-logging-${channel}`, logfile);
-    });
+    // We need to store the PID to enable killing the task.
+    // And need to parse the logfile path from stdout.
+    const stdOutCallback = async (data) => {
+      if (!investLogfile) {
+        if (`${data}`.match('Writing log messages to')) {
+          investLogfile = `${data}`.split(' ').pop().trim();
+          runningJobs[args.workspace_dir] = investRun.pid;
```

Comment:
Out of curiosity, is it possible for a user to start 2 model runs (could be different models) in the same workspace with different PIDs?  If it is, wouldn't we lose track of the PID of the first process?

Reply:
Oh thanks for catching this! I switched the key to the `jobID` that is unique to each `InvestTab` component. Since we know we can only have one running job per open tab, that should fix this.

Reply:
Ah right!  That makes a lot of sense.  Thanks!